lib/datahub-client/data_platform_catalogue/client/search.py

```diff
@@ -114,41 +104,61 @@ def search(
 
         logger.debug(json.dumps(response, indent=2))
 
-        page_results = self._parse_search_results(response)
+        # Should these 2 variables be bound or unbound?
+        page_results, malformed_result_urns = self._parse_search_results(response)
 
         return SearchResponse(
             total_results=response["total"],
             page_results=page_results,
+            malformed_result_urns=malformed_result_urns,
             facets=self._parse_facets(response.get("facets", [])),
         )
 
-    def _parse_search_results(self, response):
-        page_results = []
+    def _parse_search_results(self, response) -> Tuple[list, list]:
+        self.page_results = []
+        self.malformed_result_urns = []
         for result in response["searchResults"]:
-            entity = result["entity"]
-            entity_type = entity["type"]
-            matched_fields = self._get_matched_fields(result=result)
-
-            if entity_type == "DATASET":
-                page_results.append(
-                    self._parse_result(entity, matched_fields, ResultType.TABLE)
-                )
-            elif entity_type == "CHART":
-                page_results.append(
-                    self._parse_result(entity, matched_fields, ResultType.CHART)
-                )
-            elif entity_type == "CONTAINER":
-                page_results.append(
-                    self._parse_container(entity, matched_fields, ResultType.DATABASE)
-                )
-            elif entity_type == "DASHBOARD":
-                page_results.append(
-                    self._parse_container(entity, matched_fields, ResultType.DASHBOARD)
-                )
-            else:
-                raise ValueError(f"Unexpected entity type: {entity_type}")
-
-        return page_results
+            self._parse_result(result)
+
+        return self.page_results, self.malformed_result_urns
+
+    def _parse_result(self, result):
+        entity = result["entity"]
+        entity_type = entity["type"]
+        entity_urn = entity["urn"]
+        matched_fields = self._get_matched_fields(result=result)
+
+        if entity_type == "DATASET":
+            try:
+                parsed_result = self._parse_dataset(entity, matched_fields, ResultType.TABLE)
+                self.page_results.append(parsed_result)
+            except Exception:
```

Comment:
Minor thing, but I think we can have this try except surround the whole if/elif statement, rather than repeat it in each branch.

Reply:
good idea!